torch/_inductor/codegen/simd.py

```diff
@@ -1477,77 +1620,250 @@ def codegen_combo_kernel(self, combo_kernel_node):
+
+        # Tile either pointwise or reduction dims.
+        pointwise_ranges, reduction_ranges = node.get_ranges()
+        partial_tilings = tile_ranges(
+            is_pointwise,
+            pointwise_ranges if is_pointwise else reduction_ranges,
+            node.pointwise_or_reduction_read_writes(is_pointwise),
         )
-        deps = [
-            dep
-            for dep in itertools.chain.from_iterable(dep_sources)
-            if dep.name not in V.graph.removed_buffers and isinstance(dep, MemoryDep)
+
+        # Fill in the missing ranges.
+        full_tilings = [
+            CandidateTiling(
+                tiling=cls.complete_partial_tiling(tiling.tiling, node),
+                score=tiling.score,
+                name=tiling.name,
+            )
+            for tiling in partial_tilings
         ]
-        write_names = {dep.name for dep in rw.writes}
 
-        tilings: List[CandidateTiling] = []
+        return full_tilings
 
-        for dep in deps:
-            strides = V.graph.sizevars.stride_hints(dep.index, rw.range_vars)
-            assert len(strides) == len(ranges)
-            try:
-                split = strides.index(1) + 1
-                if split == len(ranges):
-                    continue
-                if all(s == 0 for s in strides[split:]):
-                    # if this is a broadcasted tensor and all dimensions after split are broadcast,
-                    # this is not a real split
-                    continue
+    @classmethod
+    def create_tiling(
+        cls, pw_tiling: Sequence[sympy.Expr], reduction_tiling: Sequence[sympy.Expr]
+    ) -> Dict[str, Tuple[sympy.Expr]]:
+        """
+        Create a tiling dict from pointwise and reduction splits.
+        """
+        pw_prefixes = ["z", "y", "x"][-len(pw_tiling) :]
+        reduction_prefixes = ["r0_", "r1_"][: len(reduction_tiling)]
+        return immutable_dict(
+            list(zip(pw_prefixes, pw_tiling))
+            + list(zip(reduction_prefixes, reduction_tiling))
+        )
+
+    @classmethod
+    def create_partial_tiling(
+        cls,
+        tiling: Sequence[sympy.Expr],
+        is_pointwise: bool,
+    ) -> Dict[str, sympy.Expr]:
+        return cls.create_tiling(
+            tiling if is_pointwise else [],
+            tiling if not is_pointwise else [],
+        )
 
-            except ValueError:
+    @classmethod
+    def complete_partial_tiling(
+        cls, tiling: Dict[str, sympy.Expr], node
+    ) -> Dict[str, sympy.Expr]:
+        """
+        Given a tiling for only pointwise or reduction dimensions, adds the missing one.
+        """
+        splits = list(tiling.values())
+        is_pointwise = "x" in tiling
+        missing_ranges = node.get_ranges()[1 if is_pointwise else 0]
+        missing_numel = [sympy_product(missing_ranges)]
+
+        tiling_args = (
+            (splits, missing_numel) if is_pointwise else (missing_numel, splits)
+        )
+        return cls.create_tiling(*tiling_args)
```

Comment:
I originally planned to implement joint reduction + pointwise tiling in an earlier version of this PR, e.g. [b5edad8](https://github.com/pytorch/pytorch/pull/137243/commits/b5edad86ab5bd8ce3ea41dd53f00b1884cfdad1d). But then I realized that this would require changes to the autotuning, triton codegen, and maybe kernel fusion checks as well. The PR was already getting huge, so I decided to revert that change and stick to partial tilings, to limit the blast radius.

The approach was to apply the existing partial tiling algorithm to pointwise and reduction ranges separately, then combine the partial candidates by taking the Cartesian product. For example, if we had 2 different pointwise candidates and 4 different reduction candidates, we would end up with 8 joint tiling candidates. Product tilings were scored by summing the scores of each factor. 

It would be pretty straightforward to generalize the tiling code in this PR to handle joint tilings. But given that this PR is already really big, it feels like it would be better to save that for a follow up PR. What do you think?